public_html/wp-content/mu-plugins/blocks/assets/src/shared/featured-image/index.js

```diff
@@ -0,0 +1,187 @@
+					return -1;
+				} else if ( size1.area === size2.area ) {
+					return size1.aspectRatio < size2.aspectRatio ? -1 : 1;
+				} else {
+					return 1;
+				}
+			}
+		);
+
+		this.setState(
+			{ sizes }
+		);
+	}
+
+	/**
+	 * Returns URL of appropriate image. We find appropriate by looking for an image with similar aspect ratio, and similar or greater area.
+	 * If there is no image with similar aspect ratio or area, we will return strictly bigger image then currently requested.
+	 * If there is no bigger image then currently request one, we will return the max size image URL.
+	 * Note: Core Image block allows user to select image size by name, but we cannot really do that because we will be controlling multiple images at once, and one or more images may not have all the available sizes. So an automated solution is preferred here.
+	 *
+	 * @param height
+	 * @param width
+	 */
+	getSizedUrl( height, width ) {
+		const { sizes } = this.state;
+		if ( ! sizes || sizes.length === 0 ) {
+			return "";
+		}
+
+		// Lets loop through all possible images, see if we have an image that matches requested aspect ration. if we have, we will return it. Also keep track of image which has the most closest aspect ration.
+		let smallestImage;
+		let selectedImage;
+		const requiredAspectRatio = width / height;
+
+		for ( const size of sizes ) {
+
+			// Reject all the images with less height or width then 20px.
+			if (size.height < height - 20 || size.width < width - 20) {
+				continue;
+			}
+
+			// Keep track of smallest image that is similar or bigger than our requirements. In case we won't find any image with similar aspect ratio, we will use this.
+			// Since images are sorted by size, this will be the first image that is not rejected by the size filter above.
+			if ( ! smallestImage ) {
+				smallestImage = size;
+			}
+
+			if (size.aspectRatio > requiredAspectRatio - 0.25 &&
+				size.aspectRatio < requiredAspectRatio + 0.25) {
+				// This image has similar aspect ratio to what we need. Also, since possibleImage array is sorted from lowest size, this is also the lowest size image that we can use. Lets go ahead and use this image.
+				selectedImage = size;
+				break;
+			}
+		}
+
+		if ( selectedImage ) {
+			return selectedImage.source_url;
+		} else if ( smallestImage ) {
+			return smallestImage.source_url;
+		} else {
+			// All images are small, lets just return the largest image url.
+			return sizes[ sizes.length - 1 ].source_url;
+		}
+
+	}
+
+	/**
+	 * Renders FeaturedImage component.
+	 *
+	 * @returns {*}
+	 */
+	render() {
+		const { className, alt, onChange } = this.props;
+
+		const imageSize = this.props.size || { height: 150, width: 150 };
+
+		const { height, width } = imageSize;
+
+		const imageURL = this.getSizedUrl( height, width );
+		if ( imageURL !== this.currentImageURL ) {
+			this.currentImageURL = imageURL;
+			if ( typeof( onChange ) === 'function' ) {
+				onChange( imageURL );
+			}
```

Comment:
I'm not sure I understand the purpose of this. Won't just changing the value of `src` in the `img` element cause it to update?

Also, would it make sense to store this in state instead of a custom property? Several things I've read suggest it's an anti-pattern to modify properties directly, although I couldn't tell you exactly why that is.

Reply:
This code was removed because we changed our approach. Resolving this conversation.